File: solutions/c#/Rotate List.cs
public class ListNode {
    public int val;
    public ListNode next;
    public ListNode(int val=0, ListNode next=null) {
        this.val = val;
        this.next = next;
    }
}

public class Solution {
    public ListNode RotateRight(ListNode head, int k) {
        if(head==null || head.next==null || k==0){
            return head;
        }
        ListNode temp=head;
        int len=1;
        while(temp.next!=null){
            temp=temp.next;
            len++;
        }
        temp.next=head;

        k=k%len;
        k=len-k;

        ListNode curr;
        curr=head;
        while(k> 1){
            curr=curr.next;
            k=k-1;
        }

        head=curr.next;
        curr.next=null;

        return head;

    }
}

File: solutions/python/Rotate List.py
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
class Solution:
    def rotateRight(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        if head==None or head.next==None or k==0:
            return head;
        else:
            count=1;
            temp=head;
            while temp.next:
                count=count+1;
                temp=temp.next;

            temp.next=head;
            k=k%count;
            k=count-k;
            curr=head;
            while k>1:
                curr=curr.next;
                k=k-1;
            head=curr.next
            curr.next=None;
            return head;